### PR TITLE
librbd/cache/pwl: fix bug of flush request blocked by defered IO

### PR DESCRIPTION
**Pull request by CongMinYin**
Flush requests do not need to be queued behind the defer_io queue,
should be issued immediately. Otherwise, there will be a deadlock
scenario in which dirty data is waiting for flush req, flush req is
waiting for defer_io empty, and defer_io is waiting for dirty data
persistence to release space. So this sometimes occur when the cache
is small but the IO is large or the queue depth is large.

In addition, In addition, the annotation is updated to correctly describe.

Signed-off-by: Yin Congmin <congmin.yin@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by CongMinYin** [comment]
Hi @dillaman , this pr is mainly to fix flush req blocked by defer write IO bug, it will cause IO hang when the cache is small but the IO is large or the queue depth is large.This is found in the last pr of testing VM hang in a small rwl cache.

File: src/librbd/cache/pwl/AbstractWriteLog.cc [hunk 1]
```diff
@@ -1430,6 +1432,11 @@ void AbstractWriteLog<I>::alloc_and_dispatch_io_req(C_BlockIORequestT *req)
     {
       std::lock_guard locker(m_lock);
       dispatch_here = m_deferred_ios.empty();
+      // Only flush req's total_bytes is the max uint64
+      if ((req->image_extents_summary.total_bytes
+        == std::numeric_limits<uint64_t>::max())) {
```
Comment (dillaman) [inline src/librbd/cache/pwl/AbstractWriteLog.cc:1437]: Who sets the image extent to max int?

Comment (dillaman) [inline src/librbd/cache/pwl/AbstractWriteLog.cc:1437]: Nit: move `==` to previous line and ensure indentation is appropriate

**Review by dillaman — COMMENTED** [review]
Also, small typo in the commit subject

**Review by dillaman — APPROVED** [review]
lgtm

**Pull request merged**